### PR TITLE
Add fault injection testing

### PR DESCRIPTION
**Pull request by damienomurchu**
## What
https://issues.redhat.com/browse/MGDSTRM-395
Set up fault injection testing

## Why
Test the resilience of a managed kafka service by injecting faults into the service and dependent components and assessing how it can handle unexpected events.

## How
Add resource files and make targets for the Litmus fault injection testing framework

### CODE REVIEW

**Comment by damienomurchu** [comment]
@jhellar this should be good to merge if you don't mind having a look and approving. PR just contains the pod-delete experiment at the moment, and initial CI job will just run this (we can then expand that out). 

Work for adding the CI job is still in progress and hope to look at it later today. That work is here: https://gitlab.cee.redhat.com/damurphy/app-interface/-/tree/chaos-testing-ci (I will raise a PR later)

File: install/resources/testing-chaos/chaosengines/pod-delete/chaosengine.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+apiVersion: litmuschaos.io/v1alpha1
+kind: ChaosEngine
+metadata:
+  name: pod-delete
+  namespace: kafka-cluster
```
Comment (jhellar) [inline install/resources/testing-chaos/chaosengines/pod-delete/chaosengine.yaml:5]: Should this be passed as parameter when applying the resource?

File: install/resources/testing-chaos/chaosengines/pod-delete/serviceaccount.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: pod-delete-sa
+  namespace: kafka-cluster
```
Comment (jhellar) [inline install/resources/testing-chaos/chaosengines/pod-delete/serviceaccount.yaml:6]: Same here

**Comment by jhellar** [comment]
LGTM, just namespace could be specified when resources are being applied. Also haven't tested it yet.

**Comment by damienomurchu** [comment]
Thanks @jhellar - agree with your suggestions, need to add more configurability to the targets, and will add that in a subsequent PR. 

If you're happy with that, I'll merge this now if you approve, so it won't block the CI pr that's coming shortly.

**Review by jhellar — APPROVED** [review]

**Pull request merged**